src/dropdown.ts

```diff
@@ -2304,35 +2301,17 @@ export default class Dropdown extends LitElement implements FormControl {
   }
 
   #onOptionsSelectedChange(event: Event) {
-    if (!this.multiple) {
-      for (const option of this.#optionElements) {
-        if (
-          option !== event.target &&
-          event.target instanceof DropdownOption &&
-          event.target.selected &&
-          option.selected &&
-          event.target !== this.#selectAllElementRef.value
-        ) {
-          option.selected = false;
-        }
-      }
+    if (this.#isSelectionFromSelectAllOrNone) {
+      return;
     }
 
-    if (
-      event.target !== this.#selectAllElementRef.value &&
-      !this.#isSelectionChangeFromSelectAll &&
-      this.#selectAllElementRef.value
-    ) {
-      // Update Select All to reflect the new selection or deselection.
+    if (this.#selectAllElementRef.value) {
       this.#selectAllElementRef.value.selected = this.isAllSelected;
     }
 
-    this.isShowSingleSelectIcon =
```

Comment:
Moved this inside the conditional below that contains everything else to do with single-select.